### ISSUE TITLE
Class 'str' has no '__value__' member on PEP 695 TypeAliasType

### ISSUE THREAD

**Issue by jesnie** [PyCQA/pylint#10091]
### Bug description

I have a medium-sized code-base I've been updating to PEP 695 typing, and I'm getting a lot of false positives.

Some minimal examples:

```python
class A[T]:

    def __init__(self, x: T) -> None:
        self._x = x
```

Errors with:
```
test_pylint_695_1.py:3:26: E0602: Undefined variable 'T' (undefined-variable)
```

But:
```python
type L[T] = list[T]

class A[T]:

    def __init__(self, x: T) -> None:
        self._x = x
```
Passes.

Also:
```python
def ident[T](x:T) -> T:
    return x
```
Passes

but:
```python
type L[T] = list[T]

def ident[T](x:T) -> T:
    return x
```
Fails with:
```
test_pylint_695_3.py:3:10: W0621: Redefining name 'T' from outer scope (line 1) (redefined-outer-name)
```

I suspect:
1. You're not picking up that `class A[T]` is defining the variable `T`.
2. You're not correctly scoping the `T` in `type L[T]`.

### Configuration

```ini

```

### Command used

```shell
mypy test_pylint_695* && pylint test_pylint_695*
```

### Pylint output

```python
************* Module test_pylint_695_1
test_pylint_695_1.py:3:26: E0602: Undefined variable 'T' (undefined-variable)
************* Module test_pylint_695_3
test_pylint_695_3.py:3:10: W0621: Redefining name 'T' from outer scope (line 1) (redefined-outer-name)

------------------------------------------------------------------
Your code has been rated at 5.00/10 (previous run: 5.00/10, +0.00)
```

### Expected behavior

All of these examples should pass.

### Pylint version

```shell
pylint==3.3.1
astroid==3.3.5
Python 3.12.3
```

### OS / Environment

Ubuntu 22.04.3 LTS (running under WSL)

### Additional dependencies

```python

```

**Comment by jacobtylerwalls** [comment]
Thanks for the report. We have #9335 for the first report and #9884 for the second, so I'll refocus this ticket on the 3rd issue you report here with no-member.
